app/src/main/java/com/babylon/wallet/android/di/ApplicationModule.kt

```diff
@@ -50,6 +52,21 @@ object ApplicationModule {
         return EncryptedDiskCacheClient(applicationContext, jsonSerializer)
     }
 
+    @Provides
+    @Singleton
+    fun provideStateDatabase(
+        @ApplicationContext applicationContext: Context
+    ): StateDatabase {
+        return StateDatabase.factory(applicationContext)
+    }
+
+    @Provides
+    fun provideStateDao(
```

Comment:
Should this be a singleton as well ?

Reply:
No this is exactly the same as StateApi which is constructed in a singleton okhttp client. In this case this specifc dao is constructed in a singleton database. DAOs and APIs are just object wrappers of interfaces which hold no state so there is no need to keep a single instance in memory.